### PR TITLE
feat(drivebase, intake, shooter, container, constants, calibration): …

### PR DESCRIPTION
**Pull request by TheGamer1002**
…remove setting by voltage, disable setting of Spark Max parameters in code, limit max speed of swerve, calibrate swerve modules, restructure deploy directory, change controller button and joystick bindings, add new commands for intake, shooter, and drivebase

### CODE REVIEW

**Comment by TheGamer1002** [comment]
This code has been tested and used on the real robot.

**Comment by NyxAlexandra** [comment]
I will review once off the plane :P

**Comment by TheGamer1002** [comment]
> I will review once off the plane :P

🫵 plane girl

File: src/main/java/frc/robot/FieldConstants.java [hunk 1]
```diff
@@ -20,40 +24,40 @@
...
-  public static final double FIELD_LENGTH = Units.inchesToMeters(651.223);
-  public static final double FIELD_WIDTH = Units.inchesToMeters(323.277);
-  public static final double WING_X = Units.inchesToMeters(229.201);
-  public static final double PODIUM_X = Units.inchesToMeters(126.75);
-  public static final double STARTING_LINE_X = Units.inchesToMeters(74.111);
+  public static final Measure<Distance> FIELD_LENGTH = Inches.of(651.223);
+  public static final Measure<Distance> FIELD_WIDTH = Inches.of(323.277);
+  public static final Measure<Distance> WING_X = Inches.of(229.201);
+  public static final Measure<Distance> PODIUM_X = Inches.of(126.75);
+  public static final Measure<Distance> STARTING_LINE_X = Inches.of(74.111);
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/FieldConstants.java:31]: very much like this change :D

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -45,10 +58,15 @@ public class RobotContainer
...
    * The container for the robot. Contains subsystems, OI devices, and commands.
    */
-  public RobotContainer()
-  {
-    // Configure the trigger bindings
-    configureBindings();
+  public RobotContainer() {
+    drivebase = SwerveSubsystem.getInstance();
+    intake = new Intake(new IntakeIOSparkMax());
+    shooter = new Shooter(new ShooterIOSparkFlex());
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/RobotContainer.java:64]: big fan of this change too :D

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -99,38 +120,40 @@ public RobotContainer()
...
+  public Command getAutonomousCommand() {
     // An example command will be run in autonomous
     return drivebase.getAutonomousCommand("New Auto");
   }
 
-  public void setDriveMode()
-  {
+  public void setDriveMode() {
     //drivebase.setDefaultCommand();
   }
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/RobotContainer.java:154]: will, I presume, be changed later

File: src/main/java/frc/robot/subsystems/intake/IntakeIOSparkMax.java [hunk 4]
```diff
@@ -13,36 +13,49 @@ public class IntakeIOSparkMax implements IntakeIO {
...
+//    cerealizerMotor.getPIDController().setI(Constants.IntakeConstants.I);
+//    cerealizerMotor.getPIDController().setD(Constants.IntakeConstants.D);
+//    cerealizerMotor.getPIDController().setIZone(Constants.IntakeConstants.Iz);
+//    cerealizerMotor.getPIDController().setFF(Constants.IntakeConstants.FF);
+//    cerealizerMotor.getPIDController().setOutputRange(Constants.IntakeConstants.MIN_OUTPUT, Constants.IntakeConstants.MAX_OUTPUT);
+//    cerealizerMotor.getPIDController().setSmartMotionMaxVelocity(5700, 0);
+//    cerealizerMotor.getPIDController().setSmartMotionMinOutputVelocity(0, 0);
+//
+//    intakeMotor.burnFlash();
+//    cerealizerMotor.burnFlash();
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/subsystems/intake/IntakeIOSparkMax.java:58]: dislike. git history exist for a reason. delete these comments

File: src/main/java/frc/robot/subsystems/shooter/ShooterIOSparkFlex.java [hunk 5]
```diff
@@ -16,37 +16,44 @@ public class ShooterIOSparkFlex implements ShooterIO {
...
+//    upperPID.setFF(Constants.ShooterConstants.FF);
+//
+//    lowerPID.setSmartMotionMaxVelocity(5700, 0);
+//    lowerPID.setSmartMotionMinOutputVelocity(0, 0);
+//    upperPID.setSmartMotionMaxVelocity(5700, 0);
+//    upperPID.setSmartMotionMinOutputVelocity(0, 0);
+//
+//    // Burn flash
+//    lowerMotor.burnFlash();
+//    upperMotor.burnFlash();
```
Comment (NyxAlexandra) [inline src/main/java/frc/robot/subsystems/shooter/ShooterIOSparkFlex.java:56]: same as above. delete

**Review by NyxAlexandra — DISMISSED** [review]
overall I really like these changes :D, just need to fix my nitpicks about comments

**Comment by NyxAlexandra** [comment]
for future reference, you don't need to be this granular with commit messages/pull request titles

**Review by NyxAlexandra — APPROVED** [review]
lgtm

**Comment by NyxAlexandra** [comment]
@TheGamer1002 can you fix the merge conflicts

**Comment by TheGamer1002** [comment]
There is none?
Also what does "igtm" mean?

**Pull request merged**